linkis-public-enhancements/linkis-basedata-manager/src/main/java/org/apache/linkis/basedatamanager/server/restful/UdfTreeRestfulApi.java

```diff
@@ -64,10 +73,25 @@ public Message list(
 
   @ApiOperation(value = "all", notes = "Query all data of UDF Tree", httpMethod = "GET")
   @RequestMapping(path = "/all", method = RequestMethod.GET)
-  public Message all(HttpServletRequest request, String searchName) {
+  public Message all(HttpServletRequest request, String searchName, String category) {
     ModuleUserUtils.getOperationUser(
         request, "Query all data of UDF Tree,search name:" + searchName);
-    List<UdfTreeEntity> udfTreeEntityList = udfTreeService.list();
+    List<UdfTreeEntity> udfTreeEntityList = new ArrayList<>();
+    UdfTreeEntity entity = new UdfTreeEntity();
+    entity.setCategory(category);
+    QueryWrapper<UdfTreeEntity> querySysWrapper =
+        new QueryWrapper<>(entity)
+            .eq("category", entity.getCategory())
+            .in("user_name", "sys", "expire", "share", "bdp");
```

Comment:
adjust as configuration item 
key is UDF_FUN_SYSTEM_CATEGORY
linkis.udf.fun.system.category="user_name,sys,expire,share,bdp"

Reply:
done